packages/components/inputs/text-input/src/text-input.tsx

```diff
@@ -1,127 +1,131 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import requiredIf from 'react-required-if';
+import React, { FocusEventHandler, ChangeEventHandler } from 'react';
 import { useTheme } from '@emotion/react';
-import { filterDataAttributes } from '@commercetools-uikit/utils';
+import { filterDataAttributes, warning } from '@commercetools-uikit/utils';
 import Constraints from '@commercetools-uikit/constraints';
 import { getInputStyles } from '@commercetools-uikit/input-utils';
 
-const TextInput = (props) => {
-  const theme = useTheme();
-  return (
-    <Constraints.Horizontal max={props.horizontalConstraint}>
-      <input
-        id={props.id}
-        name={props.name}
-        type="text"
-        value={props.value}
-        onChange={props.onChange}
-        onBlur={props.onBlur}
-        onFocus={props.onFocus}
-        disabled={props.isDisabled}
-        placeholder={props.placeholder}
-        readOnly={props.isReadOnly}
-        autoFocus={props.isAutofocussed}
-        autoComplete={props.autoComplete}
-        css={getInputStyles(props, theme)}
-        // Allow to override the styles by passing a `className` prop.
-        // Custom styles can also be passed using the `css` prop from emotion.
-        // https://emotion.sh/docs/css-prop#style-precedence
-        className={props.className}
-        {...filterDataAttributes(props)}
-        /* ARIA */
-        aria-readonly={props.isReadOnly}
-        role="textbox"
-        contentEditable={!props.isReadOnly}
-      />
-    </Constraints.Horizontal>
-  );
-};
-
-TextInput.displayName = 'TextInput';
-
-TextInput.propTypes = {
+type TTextInputProps = {
```

Comment:
Do we have a convention typing convention for ui-kit at the moment?

Reply:
thanks for this question, but is there something specific you're referring to?
I set a `T*` prefix to differentiate Types and Interface for devs who rely on editors without intellisense. Does that make sense?

Reply:
Basically, I'm asking if using a naming like, `T*` prefix, is a convention for ui-kit. This potentially could help with migrations for other files and avoids ambiguity.
Does that help?

Reply:
then yes it is